driver-core/src/main/java/com/datastax/driver/core/MaterializedViewMetadata.java

```diff
@@ -178,21 +172,12 @@ public String apply(ColumnMetadata input) {
         newLine(sb.append("FROM ").append(keyspaceName).append('.').append(baseTableName).append(" "), formatted);
 
         // WHERE
-        newLine(sb.append("WHERE "), formatted);
-        Iterator<ColumnMetadata> it = getPrimaryKey().iterator();
-        while(it.hasNext()) {
-            ColumnMetadata column = it.next();
-            sb.append(spaces(4, formatted)).append(Metadata.escapeId(column.getName()));
-            sb.append(" IS NOT NULL");
-            if(it.hasNext()) sb.append(" AND");
-            sb.append(" ");
-            newLine(sb, formatted);
-        }
+        newLine(sb.append("WHERE ").append(whereClause).append(' '), formatted);
```

Comment:
Are where clauses required?   The cql grammar seems to allow you to create MVs without it, but cassandra doesn't allow it.  It also seems that previously you could create MVs without it (for example from [this blog post](http://christopher-batey.blogspot.com/2015/07/cassandra-30-materialised-views-in.html) but that could be outdated).   If they are not required we should have some criteria here to omit this if the whereClause is empty or null.

I posted a question on CASSANDRA-9664.